### DISCUSSION TITLE
how to run GUI with out the flatpak

### DISCUSSION THREAD

**Discussion by zany130** [quexten/goldwarden#157] [Q&A]
flatpak is currently not working so I wanted to try running the GUI  from the terminal so I did the following

1. git clone https://github.com/quexten/goldwarden.git
2. cd goldwarden/gui
3. since there is no arch package for python-tendo I ran `python -m venv /home/zany130/.local/pip/venvs/goldwarden` (this creates a container to install python libraries this is needed because pip is disabled on arch)
4. `/home/zany130/.local/pip/venvs/goldwarden/bin/pip install tendo` to install tendo (`install -r python3-requirements.json` gives `ERROR: Invalid requirement: '{' (from line 1 of python3-requirements.json)` 
5. `/home/zany130/.local/pip/venvs/goldwarden/bin/python goldwarden_ui_main.py` to run the GUI but it fails with

```
Traceback (most recent call last):
  File "/home/zany130/Documents/GitHub/goldwarden/gui/goldwarden_ui_main.py", line 2, in <module>
    import src.linux.main as linux_main
  File "/home/zany130/Documents/GitHub/goldwarden/gui/src/linux/main.py", line 5, in <module>
    from .monitors import dbus_autofill_monitor
  File "/home/zany130/Documents/GitHub/goldwarden/gui/src/linux/monitors/dbus_autofill_monitor.py", line 4, in <module>
    from gi import require_version
ModuleNotFoundError: No module named 'gi'
```

```
inxi -b
System:
  Host: Garuda-Linux Kernel: 6.8.7-2-cachyos arch: x86_64 bits: 64
  Desktop: KDE Plasma v: 6.0.4 Distro: Garuda Linux
Machine:
  Type: Desktop Mobo: ASRock model: X470 Taichi serial: <superuser required>
    UEFI: American Megatrends v: P5.10 date: 10/20/2022
CPU:
  Info: 6-core AMD Ryzen 5 5600X [MT MCP] speed (MHz): avg: 3587
    min/max: 550/4687
Graphics:
  Device-1: AMD Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M/6850M XT]
    driver: amdgpu v: kernel
  Display: wayland server: X.org v: 1.21.1.13 with: Xwayland v: 23.2.6
    compositor: kwin_wayland driver: X: loaded: amdgpu
    unloaded: modesetting,radeon dri: radeonsi gpu: amdgpu resolution:
    1: 2048x864 2: 1396x785 3: 1536x864
  API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 24.0.5-arch1.1
    renderer: AMD Radeon RX 6700 XT (radeonsi navi22 LLVM 17.0.6 DRM 3.57
    6.8.7-2-cachyos)
Network:
  Device-1: Intel Dual Band Wireless-AC 3168NGW [Stone Peak] driver: iwlwifi
  Device-2: Intel I211 Gigabit Network driver: igb
Drives:
  Local Storage: total: 5.46 TiB used: 4.26 TiB (78.0%)
Info:
  Memory: total: 32 GiB available: 31.26 GiB used: 14.39 GiB (46.0%)
  Processes: 531 Uptime: 54m Shell: fish inxi: 3.3.34
  ```

**Comment by quexten** [comment]
The reason I only package Flatpak is because the environment is controlled. Going through all dependencies on all platforms and DE's is kind of a pain ;) 

If you want to run it locally, you need to get whatever is required for GTK4 apps. This is not in the requirements, because the Gnome flatpak runtime provides this. *maybe* this helps: https://archlinux.org/packages/extra/x86_64/gtk4/

**Reply by zany130** [reply]
Yeah, I have that, as even though I use KDE, I do use some gnome apps like gnome-firmware and gnome-disks so I *should* have all the gtk4 dep.  no idea what dependencies would be needed, but since the flatpack is working for me again, I'm not too bothered, lol. If I ever do find whats needed, Ill post here
